### PR TITLE
Compare the file names and types in target/debug

### LINKED ISSUE #41: Check for final output in `target/[release|debug]`

### PR DESCRIPTION
**Pull request by CohenArthur**
Closes #41 

This PR also slightly refactors the previous checks that were made about `target/debug/deps`

### CODE REVIEW

File: tests/test_harness.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ impl Harness {
 
         for dir_entry in dir_iter.into_iter() {
             let current_path = dir_entry?.path();
-            if current_path.extension() == extension.as_deref() {
+            if current_path.extension() == extension.as_deref() && current_path.is_file() {
```
Comment (flip1995) [inline tests/test_harness.rs]: Consider https://rust-lang.github.io/rust-clippy/master/index.html#filetype_is_file

**Review by flip1995 — COMMENTED** [review]
LGTM. One comment you may or may not address

**Review by philberty — APPROVED** [review]
LGTM nice test

File: tests/test_harness.rs (hunk 1 above)
Comment (CohenArthur) [reply tests/test_harness.rs]: Fixed in f90c6b4

**Review by flip1995 — APPROVED** [review]

**Pull request merged**